opencog/executioncontext/CMakeLists.txt

```diff
@@ -0,0 +1,22 @@
+
+# The atom_types.h file is written to the build directory
+INCLUDE_DIRECTORIES( ${CMAKE_CURRENT_BINARY_DIR})
+
+
+ADD_LIBRARY (executioncontext
```

Comment:
I think it would be cleaner to just have this in the cython directory, instead of creating a new directory for it.

I know you're thinking that the scheme bindings could use this, but that's not quite the way scheme works: it uses the concept of a "fluid", which is kind of like thread-local-storage, except that it's its a full-fledged object in it's own right.  The concept of TLS works for C++/python, because in those languages, execution is hard-coded to always be in a thread (with the threads managed by the OS kernel); roughly speaking, this is possible whenever a language has an execution stack that holds transitions from caller to callee (the C++/python stack).  In functional languages, e.g. scheme, you have calls with continuations, so when you "return" or "continue", you may return to a different thread. (its vaguely like setjmp/longjmp, but guaranteed and not dangerous)  Thus, for compatibility, in order to emulate the C++/python-style concepts of threads/TLS, scheme has the concept of "fluids".  That is where the scheme bindings search for and find the "current" atomspace.   

... and that's why this code is unlikely to be useful to anyone, except for cython, which is why I suggest moving it to the cython directory.

Reply:
I thought about that difference a bit before make the PR. Python 3.7 now also has something similar to fluids, contextvars..

Anyway If python callback will start a new thread then it won't be able to get atomspace set by PythonEval. I decided it is up to developer of the callback to decide what to do in such case..

Reply:
That sounds reasonable, I guess.